src/components/enketo-iframe.vue

```diff
@@ -102,11 +107,26 @@ function handleIframeMessage(event) {
     let eventData;
     try { eventData = JSON.parse(event.data); } catch {}
 
-    if (eventData?.enketoEvent === 'submissionsuccess' && redirectUrl.value) {
-      router.push((new URL(redirectUrl.value)).pathname);
+    if (eventData?.enketoEvent === 'submissionsuccess') {
+      if (props.actionType === 'edit') {
+        // for edit we always redirect to Submission details page
+        router.push(submissionPath(form.projectId, form.xmlFormId, props.instanceId));
+      } else if (!props.actionType && redirectUrl.value) {
+        // for public link, we read return value from query parameter. The value could be 3rd party
+        // site as well, typically a thank you page
+        try {
+          const normalizedUrl = new URL(redirectUrl.value);
```

Comment:
I think I remember us talking about doing a little validation of `normalizedUrl`. Like checking that it's not something truly strange like an ftp:// or file:// URL. Does Enketo do any validation of the URL?